torch/nn/functional.py

```diff
@@ -1789,10 +1790,10 @@ def smooth_l1_loss(input, target, size_average=None, reduce=None, reduction='ele
     See :class:`~torch.nn.SmoothL1Loss` for details.
     """
     if size_average is not None or reduce is not None:
-        reduction = _Reduction.legacy_get_enum(size_average, reduce)
-    else:
-        reduction = _Reduction.get_enum(reduction)
-    return torch._C._nn.smooth_l1_loss(input, target, reduction)
+        reduction = _Reduction.legacy_get_string(size_average, reduce)
+    return _pointwise_loss(
+        lambda a, b: torch.where(torch.abs(a - b) < 1, 0.5 * (torch.abs(a - b)) ** 2, torch.abs(a - b) - 0.5),
```

Comment:
nit: if you remove this from a lambda, you could store the result of `torch.abs(a - b)`, which would reduce the number of kernel launches by quite a bit
```python
def _smooth_l1_loss(a, b):
    t = torch.abs(a - b)
    return torch.where(t < 1, 0.5 * t ** 2, t - 0.5)
```